### PR TITLE
Add attributes to the Link to match with the data model.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opencensus/trace/Link.java [hunk 1]
```diff
@@ -44,7 +49,24 @@
    * @return a new {@code Link}.
    */
   public static Link fromSpanContext(SpanContext context, Type type) {
-    return new AutoValue_Link(context.getTraceId(), context.getSpanId(), type);
+    return new AutoValue_Link(context.getTraceId(), context.getSpanId(), type, EMPTY_ATTRIBUTES);
```
Comment (songy23) [inline api/src/main/java/io/opencensus/trace/Link.java]: s/`return fromSpanContext(context, type, EMPTY_ATTRIBUTES);` ?

Comment (bogdandrutu) [inline api/src/main/java/io/opencensus/trace/Link.java]: Unnecessary copy of the empty map.

File: api/src/main/java/io/opencensus/trace/Link.java [hunk 2]
```diff
@@ -44,7 +49,24 @@
...
+   * @param attributes the attributes of the {@code Link}.
+   * @return a new {@code Link}.
+   */
+  public static Link fromSpanContext(
+      SpanContext context, Type type, Map<String, AttributeValue> attributes) {
+    return new AutoValue_Link(
+        context.getTraceId(),
+        context.getSpanId(),
+        type,
+        Collections.unmodifiableMap(new HashMap<String, AttributeValue>(attributes)));
```
Comment (adriancole) [inline api/src/main/java/io/opencensus/trace/Link.java]: always use LinkedHashMap as iteration order differences are an unnecessary distraction

File: api/src/main/java/io/opencensus/trace/Link.java [hunk 3]
```diff
@@ -68,5 +90,12 @@ public static Link fromSpanContext(SpanContext context, Type type) {
    */
   public abstract Type getType();
 
+  /**
+   * Returns the set of attributes.
```
Comment (adriancole) [inline api/src/main/java/io/opencensus/trace/Link.java]: why you'd put attributes on a link isn't obvious. What's the use case? Can an example end up in javadoc?

**Pull request merged**